app/src/Utils/ProgressBarColours.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * @file This file contains the functions that are used to change the colour of the progress bar
+ */
+
+// type for the colours of the progress bar
+type ProgressBarColors = {
+  box1: string
+  box2: string
+  box3: string
+  bgColor: string
+}
+
+/**
+ *
+ * @param currentBox
+ * @param bgColor
+ * @returns  {ProgressBarColors} - returns the colours for the progress bar
+ */
+
+export const getProgressBarColors = (
+  currentBox: number,
+  bgColor: string,
+): ProgressBarColors => {
+  const boxes = ['#F1F1F1', '#F1F1F1', '#F1F1F1'] // default colour for all boxes while the user is on the same screen
+  for (let i = 0; i < currentBox; i++) {
+    boxes[i] = bgColor
+  }
+
+  return {
+    box1: boxes[0],
+    box2: boxes[1],
+    box3: boxes[2],
+    bgColor: boxes[3] === bgColor ? bgColor : '#F1F1F1',
```

Comment:
Tiny thing, but as good practice could use the [nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) here to simplify this to `bgColor: boxes[3] === bgColor ?? '#F1F1F1',`

(but obviously update the colour reference to call it from StaTheme)